### DISCUSSION TITLE
pgr-services goes into a crashloop due to improper db migration scripts, not able to create businessService configuration for PGR

### DISCUSSION THREAD

**Discussion by shashwat-egov** [egovernments/DIGIT-DevOps#819] [Q&A]
![pgrCrashLoopBackOff](https://user-images.githubusercontent.com/71879793/166091746-e28ac7f7-acef-4df6-b9f6-3d533dc98ca9.png)

Upon checking pgr-services logs,

![pgr-logs](https://user-images.githubusercontent.com/71879793/166092016-558e79ce-b059-475c-afec-613b5430ba22.png)

The pod is not able to find businessService config of PGR which is causing it to go into a crashloop.

Upon checking the persister service ( asynchronously inserts data into the database ) logs, 

![columnActiveDoesNotExist](https://user-images.githubusercontent.com/71879793/166091839-a1c44042-da74-4921-a194-c1a8705d7af4.png)

Seems like the db migration scripts are not in sync with the INSERT query that is getting executed to persist data onto the db. Adding this missing migration script ought to solve the issue - https://github.com/egovernments/DIGIT-Dev/blob/develop/core-services/egov-workflow-v2/src/main/resources/db/migration/main/V20210423102936__wf_alter_table_active_ddl.sql

**Comment by gajenc** [comment]
@shashwat-egov This is expected behaviour, we have already mentioned this in the document that initially the PGR-service will be in crash loop, you should run the post deployment scripts.

![image](https://user-images.githubusercontent.com/40357140/166142209-49c9d903-731d-4e7a-8b2e-e53e8097e9a5.png)

**Reply by shashwat-egov** [reply]
@gajendran-egov I followed the post deployment steps already but it did not come out of crashloop. The error does not go away. I suspect that it is pointing to persister configs which are later than v2.4 as the INSERT query getting executed seems to be out of sync with the tables of workflow service.

**Comment by shashwat-egov** [comment]
Update: This issue has been escalated to the urban team and will be fixed soon. Also, we should have a separate configs branch for quickstart too so that the queries that are getting executed and db tables do not go out of sync.
